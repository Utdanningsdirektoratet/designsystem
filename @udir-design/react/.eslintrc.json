{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*", "storybook-static"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "@udir-design/react/tsconfig.*?.json",
          "@udir-design/react/tsconfig.storybook.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
