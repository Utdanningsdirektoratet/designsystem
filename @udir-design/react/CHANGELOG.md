## 0.1.0-1 (2024-10-22)

build: don't bundle depenencies


## 0.1.0-0 (2024-10-22)

### ğŸš€ Features

- add Button component ([8d698eb](https://github.com/Utdanningsdirektoratet/designsystem/commit/8d698eb))
