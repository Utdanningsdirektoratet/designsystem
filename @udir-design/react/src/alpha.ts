/*
 * This is the entry point for the /alpha sub-module.
 * It exports the alpha-stage components, plus everything in beta (& stable).
 */
export * from './beta';
export * from './components/alpha';
export * from './utilities/hooks/alpha';
