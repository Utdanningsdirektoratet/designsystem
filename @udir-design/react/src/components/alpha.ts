/*
 * This file exports the components that are in the alpha stage of development
 */

export * from './button/Button';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './details/Details';
export * from './dialog/Dialog';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './errorSummary/ErrorSummary';
export * from './field/Field';
export * from './fieldset/Fieldset';
export * from './input/Input';
export * from './link/Link';
export * from './list/List';
export * from './loaders';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skipLink/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
export * from './textarea/Textarea';
export * from './textfield/Textfield';
export * from './toggleGroup/ToggleGroup';
export * from './tooltip/Tooltip';
export * from './typography';
