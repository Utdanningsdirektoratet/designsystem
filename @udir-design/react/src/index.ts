export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
