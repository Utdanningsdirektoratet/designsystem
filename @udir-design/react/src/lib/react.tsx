import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
