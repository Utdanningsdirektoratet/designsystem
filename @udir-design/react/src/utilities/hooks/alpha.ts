/*
 * This file exports the hooks that are in the alpha stage of development
 */

export {
  type UsePaginationProps,
  usePagination,
} from './usePagination/usePagination';
export {
  useTableOfContents,
  type useTableOfContentsProps,
} from './useTableOfContents/useTableOfContents';
