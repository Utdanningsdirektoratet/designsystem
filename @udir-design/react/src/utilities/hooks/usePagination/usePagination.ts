import {
  usePagination,
  type UsePaginationProps,
} from '@digdir/designsystemet-react';
export { usePagination, UsePaginationProps };
