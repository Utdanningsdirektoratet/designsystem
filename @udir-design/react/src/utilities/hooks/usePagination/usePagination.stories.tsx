import { useArgs } from '@storybook/preview-api';
import type { Meta, StoryObj } from '@storybook/react';
import { Pagination } from '@udir-design/react/alpha';
import { type UsePaginationProps, usePagination } from './usePagination';

const meta: Meta = {
  title: 'Utilities/usePagination',
  parameters: { chromatic: { disableSnapshot: true } },
  argTypes: {
    currentPage: {
      control: {
        type: 'number',
      },
      defaultValue: 1,
      description: 'The current page number',
      type: { name: 'number' },
    },
    totalPages: {
      control: {
        type: 'number',
      },
      defaultValue: 1,
      description: 'The total number of pages',
      type: { name: 'number' },
    },
    showPages: {
      control: {
        type: 'number',
      },
      defaultValue: 7,
      description: 'The maximum number of pages to show',
      type: { name: 'number' },
    },
    setCurrentPage: {
      description: 'Callback to set the current page',
      type: { name: 'function' },
    },
    onChange: {
      description: 'Callback when the page changes',
      type: { name: 'function' },
    },
  },
};

export default meta;

type Story = StoryObj<UsePaginationProps>;

export const Preview: Story = {
  args: {
    currentPage: 2,
    totalPages: 10,
    showPages: 7,
  },
  render(args) {
    const [, updateArgs] = useArgs();
    const { pages, nextButtonProps, prevButtonProps } = usePagination({
      ...args,
      setCurrentPage: (currentPage) => updateArgs({ currentPage }),
    });

    return (
      <Pagination aria-label="Sidenavigering">
        <Pagination.List>
          <Pagination.Item>
            <Pagination.Button aria-label="Forrige side" {...prevButtonProps}>
              Forrige
            </Pagination.Button>
          </Pagination.Item>
          {pages.map(({ page, itemKey, buttonProps }) => (
            <Pagination.Item key={itemKey}>
              {typeof page === 'number' && (
                <Pagination.Button aria-label={`Side ${page}`} {...buttonProps}>
                  {page}
                </Pagination.Button>
              )}
            </Pagination.Item>
          ))}
          <Pagination.Item>
            <Pagination.Button aria-label="Neste side" {...nextButtonProps}>
              Neste
            </Pagination.Button>
          </Pagination.Item>
        </Pagination.List>
      </Pagination>
    );
  },
};
