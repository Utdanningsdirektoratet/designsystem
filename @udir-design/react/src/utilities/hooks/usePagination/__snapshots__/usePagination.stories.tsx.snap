// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Preview 1`] = `
<nav aria-label="Sidenavigering">
  <ul>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Forrige side"
        aria-hidden="false"
      >
        Forrige
      </button>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Side 1"
      >
        1
      </button>
    </li>
    <li>
      <button
        data-variant="primary"
        type="button"
        aria-label="Side 2"
        aria-current="page"
      >
        2
      </button>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Side 3"
      >
        3
      </button>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Side 4"
      >
        4
      </button>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Side 5"
      >
        5
      </button>
    </li>
    <li>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Side 10"
      >
        10
      </button>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
        aria-label="Neste side"
        aria-hidden="false"
      >
        Neste
      </button>
    </li>
  </ul>
</nav>
`;
